contracts/StablecoinConverter.sol

```diff
@@ -260,31 +279,39 @@ contract StablecoinConverter is EpochTokenLocker {
                     currentPrices[order.buyToken],
                     currentPrices[order.sellToken]
                 );
+                buyVolume = uint128(buyVolume.mul(feeDenominator - 1) / feeDenominator);
                 revertRemainingOrder(owner, orderId, sellVolume);
                 subtractBalance(owner, tokenIdToAddressMap(order.buyToken), buyVolume);
             }
         }
     }
 
     function findPriceIndex(uint16 index, uint16[] memory tokenIdForPrice) private pure returns (uint) {
+        // return fee token
+        if (index == 0) {
+            return 0;
+        }
+        // binary search for the other tokens
         uint leftValue = 0;
         uint rightValue = tokenIdForPrice.length - 1;
         while (rightValue >= leftValue) {
             uint middleValue = leftValue + (rightValue-leftValue) / 2;
             if (tokenIdForPrice[middleValue] == index) {
-                return middleValue;
+                return middleValue + 1;
             } else if (tokenIdForPrice[middleValue] < index) {
                 leftValue = middleValue + 1;
             } else {
+                require(middleValue > 0, "Price not provided for token, underflow would have happened");
                 rightValue = middleValue - 1;
             }
         }
         revert("Price not provided for token");
     }
 
-    function checkPriceOrdering(uint16[] memory tokenIdForPrice) private pure returns (bool) {
-        for (uint i = 1; i < tokenIdForPrice.length; i++) {
-            if (tokenIdForPrice[i] <= tokenIdForPrice[i - 1]) {
+    function checkPriceOrdering(uint16[] memory tokenIdsForPrice) private pure returns (bool) {
+        require(tokenIdsForPrice[0] > 0, "price for fee token should not be overwritten");
```

Comment:
Why does being zero imply it was overwritten?

Reply:
In the price vector, the first element is reserved for the fee token. If another index is also pointing to 0, then there are two prices supplied for the fee token. The second one would might overwrite the first input then

Reply:
But as Felix suggested, I changed the message